docs/guides/computed-properties.mdx

```diff
@@ -6,40 +6,104 @@ description: 'Use object getters and setters'
 
 # Computed Properties
 
-Valtio itself doesn't provide a functionality for computed properties,
-but you can emulate it with object getters and setters.
+In Valtio you can use object & class getters and setters to create computed properties.
 
-⚠️ Note: This is a very advanced usage.
+⚠️ Note: This is advanced usage due to the nuances around `this` behavior; see the "Note about this `this`" section below.
```

Comment:
I didn't only mean about `this`. Getters are advanced in general.

Reply:
I will restore to unblock the rest of the PR, but getters are advanced? Like getters specifically in Valtio are advanced, or like fundamentally the JS/TS getter language feature itself is advanced?

Reply:
I believe getters in JS are advanced. IMO, classes (prototypes) are advanced too. Or, read "advanced" not for beginners, instead.

Reply:
Also, even though I do generally disagree (I like getters :-), I do in good faith want to hear & would like to include in the docs your rationale for how/why they are considered advanced.

Otherwise, as a newbie when I read a super-succinct statement like "This is an advanced usage <period>", my mind goes to "great...but why?" or "okay, I guess I believe you, but what specifically is the footgun I need to be careful of?".

And so besides just the direct "this is advanced" statement, I would genuinely appreciate hearing the rationale, so readers can have that insight into the "why" of your thought process, instead of having to take it at face value or guess as to the reasons.

And if it really is "getters are an advanced JS language feature", that's still good, b/c I can include that why/rationale in the docs, and users can then best evaluate if that rationale applies to them/their teams.

Thanks!

Reply:
(Sorry, didn't see your reply, that sounds good!)

Reply:
Good point. Thanks for sharing your point.

I think getters as well as proxies are advanced in JS, because it changes the default behavior. People expect plain objects to work like this.
```js
obj.foo = 2;
console.log(obj.foo) // ---> 2
```
getters and proxies can break that.

fwiw, I originally disliked proxies. Now, when I develop libraries with proxies, I pay much attention not to break JS behavior. That's one of the reason, `proxy()` in valtio doesn't have "get" handler.

JS functions are usually invoked with `fn()`. Just accessing a property invoking a function is very counter-intuitive.

In addition, valtio has notion of proxy/snapshot separation, which complicates things. state.foo can be a getter, but snap.foo won't invoke the getter...

So, I think people would need to know the clear understanding of getters and valtio. I recommend beginners to use just plain objects (no getters, no prototypes).

Reply:
> Just accessing a property invoking a function is very counter-intuitive.

Yeah, I actually do see what you mean...

I like that they hide whether "is `obj.foo` stored or calculated?" from callers, as that's an implementation detail only I should care about; but admittedly even that is a technical/non-beginner rationale.

I dunno, given getters were added to JavaScript in ES5 / 2009, I think you might be shouting into the wind in terms of taking a principled stance against them. :-)

> I recommend beginners to use just plain objects (no getters, no prototypes

Makes sense.

> fwiw, I originally disliked proxies.

Ha, yeah that is ironic given your, what, 5+ proxy-based libraries.

...maybe you will like getters someday! :-D

Reply:
well, I personally like how valtio deals object getters. I would use it, especially now that the usaeg is clear without proxyWithComputed.

But, this concept/mental model is unique (the caching is by-product of snapshot). Unless users know getters and valtio in deep, they can't really be sure how to get benefit.

Last but not least, you are far smarter than the average.